circus-train-aws-sns/README.md

```diff
@@ -53,28 +70,110 @@ Currently the S3 credentials specified in the JCEKS key store are used to authen
       security:
         credential-provider: jceks://hdfs[@namenode:port]/hcom/etl/<team>/conf/aws-<account>.jceks
 
-## Example message
+## JSON Messages
+The following table describes all the fields that may be present in the JSON message that is sent to the SNS 
+topic.
+
+|Field Name|Type|When present|Description|
+|----|----|----|----|
+|`protocolVersion`|String|Always|The [semantic version number](https://semver.org/) of the message|
+|`type`|String Enum Value|Always|One of: START, SUCCESS or FAILURE| 
+|`headers`|Map|When configured|Any headers that were set via the `sns-event-listener.headers` configuration property|
+|`startTime`|String |Always|Time when the Circus Train job started. Formatted as a date time in ISO8601 format (yyyy-MM-ddTHH:mm:ss.SSSZZ)|
+|`endTime`|String |Only when `type` is SUCCESS or FAILURE|Time when the Circus Train job ended. Formatted as a date time in ISO8601 format (yyyy-MM-ddTHH:mm:ss.SSSZZ)|
+|`eventId`|String|Always|The Circus Train event id|
+|`sourceCatalog`|String|Only when configured|The configured name of the source catalog|
+|`replicaCatalog`|String|Only when configured|The configured name of the replicat catalog|
+|`sourceTable`|String|Always|The fully qualified name of the source table|
+|`replicaTable`|String|Always|The fully qualified name of the replica table|
+|`replicaTableLocation`|String|Always|The replicate table location|
+|`replicaMetastoreUris`|String|Always|The Hive metastore URIs of the replica Hive Thrift metastore|
+|`partitionKeys`|Ordered Map of String->String|Only on successful replication of a partitioned table|A map where the keys are the partition key names and the values are the partition key types, the order output in the JSON matches the order of the values in the `modifiedPartitions` field below|
+|`modifiedPartitions`|List of List of Strings|Only on successful replication of a partitioned table|A list containing Lists of string representing the partition key values|
+|`bytesReplicated`|Long|Only on replication of `type` SUCCESS of a table containing data|The number of bytes of data replicated|
+|`errorMessage`|String|Only on replication of `type` FAILURE|A message describing the cause of the failure|
```

Comment:
what actually happens if the message is too big for SNS to process, does it get truncated or fails to sent?

Reply:
Ok just read the Limitations bit that explains what happens.

Reply:
That's a good question. I didn't change the sending code, if you look at the publish() method in SNSListener you'll see that we log a warning and then we try send it any way but capture the exception. My guess is that the AWS client/server will throw an exception. I'll take a look a bit later, would be nice to know.